index/scorch/segment/zap/contentcoder.go

```diff
@@ -146,33 +161,55 @@ func (c *chunkedContentCoder) Add(docNum uint64, vals []byte) error {
 }
 
 // Write commits all the encoded chunked contents to the provided writer.
-func (c *chunkedContentCoder) Write(w io.Writer) (int, error) {
+//
+// | ..... data ..... | chunk offsets (varints)
+// | position of chunk offsets (uint64) | number of offsets (uint64) |
+//
+func (c *chunkedContentCoder) Write() (int, error) {
 	var tw int
-	buf := make([]byte, binary.MaxVarintLen64)
-	// write out the number of chunks
-	n := binary.PutUvarint(buf, uint64(len(c.chunkLens)))
-	nw, err := w.Write(buf[:n])
-	tw += nw
-	if err != nil {
-		return tw, err
+
+	if c.final != nil {
+		// write out the data section first
+		nw, err := c.w.Write(c.final)
+		tw += nw
+		if err != nil {
+			return tw, err
+		}
 	}
 
+	chunkOffsetsStart := uint64(tw)
+
+	buf := make([]byte, binary.MaxVarintLen64)
```

Comment:
instead of allocating new memory here, you ought to be done with c.final, so probably can reuse it and reslice it (but need to check its cap() first)

Reply:
Done.